sdk/core/azure-core/test/ut/datetime.cpp

```diff
@@ -765,3 +765,16 @@ TEST(DateTime, TimeRoundtrip)
   TestDateTimeRoundtrip<DateTime::TimeFractionFormat::AllDigits>("2021-02-05T10:00:00.0000000Z");
   TestDateTimeRoundtrip<DateTime::TimeFractionFormat::AllDigits>("2021-02-05T20:00:00.0000000Z");
 }
+
+TEST(DateTime, ToStringNoArg)
+{
+  auto dt = DateTime::Parse("2013-05-17T01:02:03.1230000Z", DateTime::DateFormat::Rfc3339);
+  EXPECT_EQ(dt.ToString(), "2013-05-17T01:02:03.123Z");
+}
+
+TEST(DateTime, ToStringOneArg)
+{
+  auto dt = DateTime::Parse("2013-05-17T01:02:03.1230000Z", DateTime::DateFormat::Rfc3339);
+  EXPECT_EQ(dt.ToString(DateTime::DateFormat::Rfc3339), "2013-05-17T01:02:03.123Z");
+  EXPECT_EQ(dt.ToString(DateTime::DateFormat::Rfc1123), "Fri, 17 May 2013 01:02:03 GMT");
```

Comment:
The  issue with this approach is that the caller is now passing in an unnecessary (and potentially invalid) `TimeFractionFormat` value which doesn't apply to Rfc1123. They might now try to pass in different values which won't change the output at all which will be a source of confusion.

As an aside, there is a general concern that folks don't typically realize optional parameters are optional, and still try to fill them in, which makes the call-site quite cumbersome. Having method overloads simplifies that.

Reply:
Yes, that's why there used to be `ToRfc3339String()` - we've thought that it would be confusing to have a parameter that has no effect, if Rfc1123 is used (because there are not fractional seconds in RFC 1123). But, it turns out, having two ToString() methods is more confusing, so during the API review we were asked to merge them both into a single method, even though one parameter would have no effect, and everyone agreed.

With your change, it is not possible to write `dt.ToString(DateTime::DateFormat::Rfc1123)`.